### DISCUSSION TITLE
How to properly handle 'responseType' in an adapter ?

### DISCUSSION THREAD

**Discussion by KaKi87** [axios/axios#5204] [Q&A]
Hello,

The README says that [the default value for `config.responseType` is `json`](https://github.com/axios/axios/blob/v1.x/README.md?plain=1#L408).

Which is true, since `response.data` is an object, when the server replies with JSON.

But, `response.data` is also a string, when `config.responseType` is either `undefined` or `json`, despite the server replying with anything but JSON.

That behavior is welcome, but I don't see where in either of [`http`](https://github.com/axios/axios/blob/v1.x/lib/adapters/http.js) or [`xhr`](https://github.com/axios/axios/blob/v1.x/lib/adapters/xhr.js) adapters this is specified ?

As I'm writing [an adapter for Tauri](https://git.kaki87.net/KaKi87/axios-tauri-adapter) and forgot to reproduce this behavior, I'd like to know exactly is it made in official adapters, and if there's other logic I should know about, in order to be as accurate as possible to the original `axios` experience.

Additionally, I think it would be interesting to make dev-oriented resources on how to make adapters in general (especially for preserving the original features and experience), as I found none and had to improvise it, like anyone who ever done that I guess.

Thanks

**Comment by jasonsaayman** [comment]
please check the latest version as this should not be the case. should this still be happening please open a new issue and not a discussion

**Reply by KaKi87** [reply]
I was not reporting an issue here, I was asking how to reproduce behavior. If the behavior changed, then I would like to know what is the new behavior and how to reproduce it as well. My question and request remain unanswered. Thank you

**Reply by jasonsaayman** [reply]
we should now be sending json always if the string is json like / or parseable

**Reply by jasonsaayman** [reply]
i would suggest following the code paths in the response transforms for more information if i re-collect correctly
